qemu/tests/qemu_guest_agent.py

```diff
@@ -1849,6 +1853,124 @@ def gagent_check_with_migrate(self, test, params, env):
         error_context.context("Verify if guest agent works.", logging.info)
         self.gagent_verify(self.params, self.vm)
 
+    @error_context.context_aware
+    def gagent_check_umount_frozen(self, test, params, env):
+        """
+        Umount file system while fs freeze.
+
+        Steps:
+        1) boot guest with a new data storage.
+        2) format disk and mount it.
+        3) freeze fs.
+        4) umount fs/offline the volume in guest
+        5) thaw fs.
+        6) mount fs and online it again.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        def wrap_windows_cmd(cmd):
+            """
+            add header and footer for cmd in order to run it in diskpart tool.
+
+            :param cmd: cmd to be wrapped.
+            :return: wrapped cmd
+            """
+            disk = "disk_" + ''.join(random.sample(string.ascii_letters +
+                                                   string.digits, 4))
+            cmd_header = "echo list disk > " + disk
+            cmd_header += " && echo select disk %s >> " + disk
+            cmd_footer = " echo exit >> " + disk
+            cmd_footer += " && diskpart /s " + disk
+            cmd_footer += " && del /f " + disk
+            cmd_list = []
+            for i in cmd.split(";"):
+                i += " >> " + disk
+                cmd_list.append(i)
+            cmd = " && ".join(cmd_list)
```

Comment:
`cmd = cmd.replace(';', '>> %s &&' disk) + '>> %s' % disk`? While actually I see no where has `;` for `cmd`, so directly `cmd += " >> %s" % disk` is enough?

Reply:
You are right, modified